### DISCUSSION TITLE
accelerate core process

### DISCUSSION THREAD

**Discussion by jeaunrg** [Hboni/francis#31] [Ideas]
There is an mri tool called cupyimg (found in https://github.com/mritools/cupyimg) that accelerate image processing using CUDA, and cupy. 
From cupyimg we can access many skimage functions that we've already used. 
- an erosion on (256, 256, 256) numpy image with skimage takes 6s
- the same erosion with a cupy image and cupyimg.skimage takes 1s

Maybe we should try to use it for our processes? What do you think?

**Comment by Hboni** [comment]
It seems interesting and be useful for further version, but before implementing with GPU is not possible to search to make parallel processing, like [Dask](https://dask.org/). I already heard about it but never use it.

**Reply by jeaunrg** [reply]
Never heard about it but it seems even better than cupy, good one

**Reply by jeaunrg** [reply]
I checked the dask optimization.
In fact we save time on the process itself but we loose even more time with dask to numpy transformation.*

_Example:_
numpy addition process: 1.2s
with dask: 0.008s
**But** 
converting numpy to dask (`from_array`): 7s 
converting dask to numpy (`compute`): 29s

**Reply by Hboni** [reply]
Ok, I heard a lot of dask for parallel computing big operations. For us maybe it is not relevant because our operations are not too long 😅 
I think that subject will not b resolved easily. There is a note in the project board, I think we can remove it and create an issue for that?

**Reply by jeaunrg** [reply]
If you are talking about https://github.com/Hboni/francis/projects/1#card-56459262 its not the same issue there,
when I create this card I was thinking about putting heavy processes in the background with QThread. 
Once again you should have a look at my demo app https://github.com/jeaunrg/ModelViewPresenter_app which handle model process inside QThread --> no app freeze while processing! very useful

**Reply by Hboni** [reply]
Oh sorry, I thought it was linked. My bad 😓

But I think that both acceleration of core processes and avoiding freeze must be corrected in a next release. When the ideas are ready it can be good to create issues with good tags 😉
